Add axes ordering docs

stellaprins opened the pull request:
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
To help users check which axis is which (0, 1, or 2). This is needed to correctly determine the anatomical orientation of their data.

**What does this PR do?**
- Adds section on how to unambiguously determine axis order using the voxel coordinate in the napari status bar.
- Fixes a mistake I caught orientation section

## References
Resolves #402 

## How has this PR been tested?
I built the docs locally and visually inspected them
<img width="558" height="550" alt="Image" src="https://github.com/user-attachments/assets/e07103af-3265-4279-bd4a-eb915b1f331e" />

<img width="562" height="604" alt="Image" src="https://github.com/user-attachments/assets/b0cf6603-48ad-43b0-8c5d-3772e8955857" />

## Is this a breaking change?
Nope

## Does this PR require an update to the documentation?

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

stellaprins commented:
Let me know if this looks helpful, or does it feel confusing or unnecessarily detailed? Any feedback is very much appreciated.

Also, in FIJI, I could not quite figure out how to unambiguously check which displayed axis corresponds to which dimension in the underlying data array. Do you know how to do this @K-Meech

K-Meech commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -42,7 +90,7 @@ If the first image in the stack corresponds to the bottom of the brain, the orig
 <img src="images/sagittal.png" alt="sagittal" height="143"/>
 
-If the stack is sliced from the left side of the brain toward the right, the origin corresponds to the most right (`r`), superior (`s`), posterior (`p`) part of the brain, so the origin string is `rsp`.
+If the stack is sliced from the left side of the brain toward the right, the origin corresponds to the most left (`l`), superior (`s`), posterior (`p`) part of the brain, so the origin string is `lsp`.
```
It's quite possible I am misunderstanding (!), but wouldn't this be `lsa` orientation? The sagittal image shown here is labelled above as the top left pixel being anterior and superior (rather than posterior?)

Plus, below in the voxel sizes section it is referred to as 'as above, the origin of your data (first, top left voxel) is the most anterior, superior, left part of the brain'

K-Meech commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -2,8 +2,54 @@
+- Axis 2: Slice depth
+
+### FIJI
+By default, FIJI opens image stacks so that the x-axis corresponds to axis 2, the y-axis corresponds to axis 1, and the z-axis corresponds axis 0.
```
```suggestion
By default, FIJI opens image stacks so that the x-axis (image width) corresponds to axis 2, the y-axis (image height) corresponds to axis 1, and the z-axis (slice depth) corresponds to axis 0.
```
Maybe relate it back to height / width / depth as you did above?

K-Meech commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -2,8 +2,54 @@
+By default, FIJI opens image stacks so that the x-axis corresponds to axis 2, the y-axis corresponds to axis 1, and the z-axis corresponds axis 0.
+
+:::{note}
+FIJI displays x, y, z coordinates in the status bar when you hover over the image. Unlike napari, in FIJI these refer to the display, not the underlying data structure. This means FIJI’s axis labels do not necessarily match the numeric axes (0, 1, 2) used by BrainGlobe, and these coordinates cannot be used to determine which axis is which when the display of the axes has been rearranged.
```
Maybe be a bit more explicit here by giving an example of when it doesn't correspond? e.g.

FIJI displays x, y, z coordinates in the status bar when you hover over the image. In the default view, this corresponds to the numeric axes (2, 1, 0), but this isn't the case after running various Fiji commands. For example, opening `Image > Stacks > Orthogonal Views` and hovering over the displayed images, shows the x coordinate is always image width and the y coordinate is always image height, even though the image has a different orientation in each view. Fiji's axis labels refer to display, rather than the underlying data structure as in napari - so make sure you are using the default view to determin which axis is which, and none of Fiji's views that display these in different orienations.

stellaprins replied:
Changed this to, referring to the Fiji documentation for Fiji specific operations

> 
> ### Fiji
> By default, Fiji opens image stacks so that the x-axis corresponds to axis 2 (image width), the y-axis corresponds to axis 1 (image height), and the z-axis corresponds axis 0 (image depth). The x, y, z coordinates in the status bar when you hover over the image. 
> 
> Unlike napari, in Fiji these refer to the display, not the underlying data structure. You can for example see this by using [Orthogonal Views](https://imagej.net/imaging/z-functions#orthogonal-views) to visualise 3D data. Each panel shows a different orientation of the image. However, hovering over the displayed images, shows the x coordinate is always image width and the y coordinate is always image height.
> 
> This means Fiji’s axis labels do not necessarily match the numeric axes (0, 1, 2) used by BrainGlobe, and these coordinates cannot be used to determine which axis is which when the display of the axes has been rearranged. So when using Fiji, make sure you are using the default view to determine which axis is which.

adamltyson commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -2,8 +2,54 @@
+- Axis 2: Image width
+
+:::{note}
+BrainGlobe uses axis labels 0, 1, 2, instead of z, y, x. This is because the underlying data is stored as NumPy arrays, which use numeric indices for their dimensions. So the axis labels 0, 1, 2 refer directly to the structure of the underlying data and ensure our documentation matches how the data is represented. 
```
Not sure about this. The reason what we don't use x,y,z is mostly because people don't necessarily agree with what x,y,z is. E.g. often people assume anterior-posterior is Z, even if they acquired the data horizontally.

Perhaps your version is good though, as it's explanation without explicitly saying "its because mistakes are common".

stellaprins replied:
Added a slightly altered version of your suggestion @alessandrofelder:
> BrainGlobe uses axis labels 0, 1, 2, instead of z, y, x. This is because the underlying data is stored as NumPy arrays, which use numeric indices for their dimensions. **Mapping to z, y, x can lead to ambiguity because different mapping conventions exist. To avoid this, we label axes 0, 1, 2 referring** directly to the structure of the underlying data and ensure our documentation matches how the data is represented.

stellaprins commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -2,8 +2,54 @@
+Misinterpreting axes can lead to incorrect atlas generation or analysis. So before you can specify the orientation and resolution of your data, it is a good idea to figure out which axis maps on to which dimension in the underlying data.
+
+When you open an image in napari or FIJI, the default ordering of the axes is:
+- Axis 0: Slice depth 
```
Good point. I like "image depth".

stellaprins commented:
Thanks @K-Meech and @adamltyson for the comments! This is helpful.

> I'm slightly concerned that there's so much info here it may be confusing. 

Yes. I was getting worried about this too when I was writing this. Happy to cut out parts or even most of it if it is uneccessarily confusing.

stellaprins commented:
Just visually inspected the built docs and it all looks fine. 

Some screenshots of the things I changed after the first review round:

# Updated "note" and added diagrams (indicating how images open by default in napari and Fiji)

<img width="631" height="797" alt="image" src="https://github.com/user-attachments/assets/60543ce5-dbbc-456c-a7f4-a1dfdc3efae6" />


# Added infinitely looping gifs to Handedness section

<img width="626" height="650" alt="image" src="https://github.com/user-attachments/assets/f1634bb0-f8b7-4ee0-8170-15ca3fd6f958" />

alessandrofelder commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -2,8 +2,58 @@
+| Moving the cursor horizontally (width) |[0 1 **2**]          | 
+
+### Fiji
+By default, Fiji opens image stacks so that the x-axis corresponds to axis 2 (image width), the y-axis corresponds to axis 1 (image height), and the z-axis corresponds axis 0 (image depth). The x, y, z coordinates in the status bar when you hover over the image. 
```
```suggestion
By default, Fiji opens image stacks so that the x-axis corresponds to axis 2 (image width), the y-axis corresponds to axis 1 (image height), and the z-axis corresponds axis 0 (image depth). The x, y, z coordinates are displayed in the status bar when you hover over the image. 
```

alessandrofelder commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -42,19 +94,29 @@ If the first image in the stack corresponds to the bottom of the brain, the orig
 
 As an example, we will assume, as above, the origin of your data (first, top left voxel) is the most anterior, superior, 
-left part of the brain. If your plane spacing (i.e. the slice depth) is 5 microns, and your in-plane resolution is 2x2 
+left part of the brain. If your plane spacing (i.e. the image depth) is 5 microns, and your in-plane resolution is 2x2 
```
```suggestion
left part of the brain. If your plane spacing (i.e. the slice depth) is 5 microns, and your in-plane resolution is 2x2 
```

I would say image depth would be the depth of the whole image stack, but here we mean the depth between two consecutive slices of the image.

alessandrofelder commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -42,19 +94,29 @@ If the first image in the stack corresponds to the bottom of the brain, the orig
+|![left handed](images/left_handed.gif)  | ![right-handed](images/right_handed.gif)|
+
+
+*3D render of 25 μm mouse brain atlas by [Wang et al., 2020](https://doi.org/10.1016/j.cell.2020.04.007), visualised in napari with visible axes, using the [brainrender-napari plugin](https://napari-hub.org/plugins/brainrender-napari.html)*.
```
```suggestion
*3D render of 25 μm mouse brain atlas by [Wang et al., 2020](https://doi.org/10.1016/j.cell.2020.04.007), visualised in napari with visible axes at the origin, using the [brainrender-napari plugin](https://napari-hub.org/plugins/brainrender-napari.html). Like other BrainGlobe atlases, its orientation is `ASR`. The `R` means that the right hemisphere is always closer to the origin, and the right hemisphere looks like it's a left hemisphere in the right-handed coordinate system.*
```

Suggestions for being super explicit.

stellaprins replied:
Yes. I think this is very useful. I changed the last sentence slightly (but feel free to change back if you think that to how you prefer before merging!)

In bold the parts I've changed
> *The `R` means that the right hemisphere is **always** closer to the origin, **and the right hemisphere looks like it's a left hemisphere in the right-handed coordinate system**.*

my suggestion:
> *The `R` means that the right hemisphere is closer to the origin. Note that in the right-handed coordinate system, the right hemisphere looks like it's a left hemisphere.*

alessandrofelder commented on docs/source/documentation/setting-up/image-definition.md:
```diff
@@ -55,6 +107,16 @@ microns, then the voxel sizes would be `5 2 2`.
+|![left handed](images/left_handed.gif)  | ![right-handed](images/right_handed.gif)|
+
+
+*3D render of 25 μm mouse brain atlas by [Wang et al., 2020](https://doi.org/10.1016/j.cell.2020.04.007), visualised in napari with visible axes at the origin, using the [brainrender-napari plugin](https://napari-hub.org/plugins/brainrender-napari.html). Like other BrainGlobe atlases, its orientation is `ASR`. The `R` means that the right hemisphere is closer to the origin. Note that in the right ...
```
```suggestion
*3D render of 25 μm mouse brain atlas by [Wang et al., 2020](https://doi.org/10.1016/j.cell.2020.04.007), visualised in napari with visible axes at the origin, using the [brainrender-napari plugin](https://napari-hub.org/plugins/brainrender-napari.html). Like other BrainGlobe atlases, its orientation is `ASR`. The `R` means that the right hemisphere is closer to the origin. Therefore, in the right-handed coordinate system, the right hemisphere looks like it's a left hemisphere.*
```
Small tweak to emphasize the relationship between the last two sentences
